src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AdditionalDesignTimeBuildInputDataSource.cs

```diff
@@ -1,71 +1,26 @@
 
-using System;
-using System.Collections.Immutable;
 using System.ComponentModel.Composition;
-using System.Linq;
-using System.Threading.Tasks.Dataflow;
-using Microsoft.VisualStudio.ProjectSystem.Properties;
 
 namespace Microsoft.VisualStudio.ProjectSystem
 {
     /// <summary>
     ///     Responsible for producing AdditionalDesignTimeBuildInput items (such as project.assets.json)
-    ///     so that they can be watched, and when changed trigger re-evaluation.
+    ///     so that they can be watched, and when changed, trigger re-evaluation.
     /// </summary>
     [Export(typeof(IFileWatchDataSource))]
     [AppliesTo(ProjectCapability.PackageReferences)]
-    internal class AdditionalDesignTimeBuildInputDataSource : ChainedProjectValueDataSourceBase<FileWatchData>, IFileWatchDataSource
+    internal class AdditionalDesignTimeBuildInputDataSource : AbstractItemFileWatchDataSource
     {
-        private readonly ConfiguredProject _project;
-        private readonly IProjectSubscriptionService _projectSubscriptionService;
-
         [ImportingConstructor]
         public AdditionalDesignTimeBuildInputDataSource(ConfiguredProject project, IProjectSubscriptionService projectSubscriptionService)
-            : base(project.Services, synchronousDisposal:true, registerDataSource: false)
-        {
-            _project = project;
-            _projectSubscriptionService = projectSubscriptionService;
-        }
-
-        protected override UnconfiguredProject ContainingProject
+            : base(project, projectSubscriptionService)
         {
-            get { return _project.UnconfiguredProject; }
         }
 
-        protected override IDisposable LinkExternalInput(ITargetBlock<IProjectVersionedValue<FileWatchData>> targetBlock)
-        {
-            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _projectSubscriptionService.ProjectRuleSource;
-
-            // Transform the changes from evaluation/design-time build -> restore data
-            DisposableValue<ISourceBlock<IProjectVersionedValue<FileWatchData>>> transformBlock = source.SourceBlock
-                                                                                                        .TransformWithNoDelta(update => update.Derive(u => CreateFileWatch(u.CurrentState)),
-                                                                                                            suppressVersionOnlyUpdates: true,
-                                                                                                            ruleNames: new[] { AdditionalDesignTimeBuildInput.SchemaName });
-
-            // Set the link up so that we publish changes to target block
-            transformBlock.Value.LinkTo(targetBlock, DataflowOption.PropagateCompletion);
+        public override string ItemRuleName => AdditionalDesignTimeBuildInput.SchemaName;
```

Comment:
Perhaps this property should be renamed `SchemaName` for consistency?

Reply:
It's RuleName everywhere else in CPS, the rule itself is inconsistent.

Reply:
But meh, I've updated it along with FullPath so its consistent with the rule.